src/test/java/fi/fmi/avi/archiver/message/populator/StationIcaoCodeReplacerTest.java

```diff
@@ -0,0 +1,38 @@
+package fi.fmi.avi.archiver.message.populator;
+
+import fi.fmi.avi.archiver.file.InputAviationMessage;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class StationIcaoCodeReplacerTest {
```

Comment:
Any need for testing use of backreferences? E.g. `^YU.(.)$' to `XXX$1`.

Reply:
I added a test case.